src/DetectSecretsLinter.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * Copyright 2020 Pinterest, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Detect secrets in code base. 
+ */
+final class DetectSecretsLinter extends PythonExternalLinter {
+
+  private $message = "";
+
+  public function getInfoName() {
+      return 'detect-secrets'; 
+  }
+
+  public function getInfoURI() {
+      return 'https://github.com/Yelp/detect-secrets';
+  }
+
+  public function getInfoDescription() {
+      return 'Detect potential secrets in files to prevent accidental commit';
+  }
+
+  public function getLinterName() {
+      return 'detect-secrets';
+  }
+
+  public function getLinterConfigurationName() {
+      return 'detect-secrets';
+  }
+
+  public function getLinterConfigurationOptions() {
+      $options = array(
+       'detect-secrets.message' => array(
+         'type' => 'optional string',
+         'help' => $this->message,
```

Comment:
Here you should provide a help string for users to understand the purpose of this configuration option. Wrap your text in the `pht()` function so that it will format nicely for console output.